circom/tests/simple/blocks_01.circom

```diff
@@ -0,0 +1,56 @@
+// REQUIRES: circom
+// RUN: rm -rf %t && mkdir %t && %circom --llzk -o %t %s | sed -n 's/.*Written successfully:.* \(.*\)/\1/p' | xargs cat | FileCheck %s --enable-var-scope
+// END.
+
+pragma circom 2.0.0;
+
+// This test demonstrates that variables defined outside a block can be overwritten
+// inside blocks because no "error[T3001]: False assert reached" is raised but if any
+// of the asserts are changed, the error will be raised.
+template B() {
+  var x = 5;
+  {
+    assert(x == 5);
+    x = 10; // overwrites value of x
+    assert(x == 10);
+  }
+  assert(x == 10);
+}
+
+component main = B();
+
+// CHECK-LABEL: module attributes {veridise.lang = "llzk"} {
+// CHECK-LABEL:   struct.def @B<[]> {
+// CHECK-LABEL:     function.def @compute
+// CHECK-SAME:      () -> !struct.type<@B<[]>> attributes {function.allow_witness} {
+// CHECK-NEXT:        %[[VAL_0:[0-9a-zA-Z_\.]+]] = struct.new : <@B<[]>>
+// CHECK-NEXT:        %[[VAL_1:[0-9a-zA-Z_\.]+]] = felt.const  5
+// CHECK-NEXT:        %[[VAL_2:[0-9a-zA-Z_\.]+]] = felt.const  5
+// CHECK-NEXT:        %[[VAL_3:[0-9a-zA-Z_\.]+]] = bool.cmp eq(%[[VAL_1]], %[[VAL_2]])
+// CHECK-NEXT:        bool.assert %[[VAL_3]], "assertion failed"
```

Comment:
Not in scope for this PR but it could be interesting to augment these messages with debugging information; `assertion failed in <file>:<line>:<column>: <expr>` where `expr` is the circom code inside `assert(<expr>)`